adam_core/coordinates/tests/test_transforms_translation.py

```diff
@@ -0,0 +1,146 @@
+import numpy as np
+from astropy import units as u
+from astropy.time import Time
+from quivr.concat import concatenate
+
+from ...constants import Constants as c
+from ..cartesian import CartesianCoordinates
+from ..origin import Origin, OriginCodes
+from ..times import Times
+from ..transform import cartesian_to_origin
+
+MU = c.MU
+
+
+def test_cartesian_to_origin(orbital_elements, orbital_elements_barycentric):
+    # Test cartesian_to_origin correctly converts between heliocentric and
+    # barycentric cartesian coordinates
+
+    cartesian_coordinates_heliocentric = CartesianCoordinates.from_kwargs(
+        x=orbital_elements["x"].values,
+        y=orbital_elements["y"].values,
+        z=orbital_elements["z"].values,
+        vx=orbital_elements["vx"].values,
+        vy=orbital_elements["vy"].values,
+        vz=orbital_elements["vz"].values,
+        time=Times.from_astropy(
+            Time(orbital_elements["mjd_tdb"].values, format="mjd", scale="tdb")
+        ),
+        frame="ecliptic",
+        origin=Origin.from_kwargs(code=["SUN" for i in range(len(orbital_elements))]),
+    )
+
+    cartesian_coordinates_barycentric = CartesianCoordinates.from_kwargs(
+        x=orbital_elements_barycentric["x"].values,
+        y=orbital_elements_barycentric["y"].values,
+        z=orbital_elements_barycentric["z"].values,
+        vx=orbital_elements_barycentric["vx"].values,
+        vy=orbital_elements_barycentric["vy"].values,
+        vz=orbital_elements_barycentric["vz"].values,
+        time=Times.from_astropy(
+            Time(
+                orbital_elements_barycentric["mjd_tdb"].values,
+                format="mjd",
+                scale="tdb",
+            )
+        ),
+        frame="ecliptic",
+        origin=Origin.from_kwargs(
+            code=[
+                "SOLAR_SYSTEM_BARYCENTER"
```

Comment:
`OriginCodes.SOLAR_SYSTEM_BARYCENTER` has integer values corresponding to NAIF IDs. `str(OriginCodes.SOLAR_SYSTEM_BARYCENTER)` returns `"OriginCodes.SOLAR_SYSTEM_BARYCENTER"` (which is weird). I think broadly it would be nice if we could drop the strings in favor of storing integers but I'm worried we will lose readability. What do you think?

Reply:
Hm, I'm not sure. You could add a `__str__` method to `OriginCodes` that does what you want - returns `self.name`.

or you could do `OriginCodes.SOLAR_SYSTEM_BARYCENTER.name` as the literal?

I'm not really convinced these are better in any way. They might be merely different without any actual effect. I think what you have is fine, unless you feel inspired to really dig in.